pkg/cmd/template/data_values_flags.go

```diff
@@ -242,23 +296,26 @@ func (s *DataValuesFlags) buildOverlay(keyPieces []string, value interface{}, de
 	pos.SetFile(fmt.Sprintf("key '%s' (%s)", strings.Join(keyPieces, dvsMapKeySep), desc))
 
 	for _, piece := range keyPieces {
-		newMap := &yamlmeta.Map{}
-		nodeAnns := template.NodeAnnotations{}
-
-		if strings.HasSuffix(piece, missingOkSuffix) {
+		// In previous versions of ytt '+' key suffix was meant to indicate
+		// "@overlay/match missing_ok=True". It's no longer needed as all keys
+		// are considered as such. Data values schemas should be enough to
+		// provide key checking/validations.
+		// (Stripped for backwards compatibility.)
```

Comment:
This feature was not documented; few if any users knew about it.

What do you think of simply removing it, reducing code complexity?

Reply:
i dont have a strong feeling. could go either way. any preference?

Reply:
Personally, I'd delete it (and we'll note the breaking change) except: keep the line about "schemas should be enough to provide key checking/validations." as a kind of justification as to why these inputs are so permissive.

I don't see real value in keeping the code or reference to how things "were."

I _do_ see value in making explicit an assumption one part of the code holds about another's responsibilities to help tie together how we expect the overall experience to unfold.

Reply:
ok.

Reply:
done.